sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -50,161 +50,162 @@
 #define TIMEOUT_MQTT_RECEIVE_MS (60 * 1000)
 #define TIMEOUT_MQTT_DISCONNECT_MS (10 * 1000)
 
-static char mqtt_client_id[128];
-static char mqtt_username[128];
-static char register_publish_topic[128];
-static char query_topic[256];
-static char global_provisioning_endpoint[256] = { 0 };
-static char id_scope[16] = { 0 };
-static char registration_id[256] = { 0 };
-static char x509_cert_pem_file[256] = { 0 };
-static char x509_trust_pem_file[256] = { 0 };
+// Logging
+#define LOG_ERROR(...) \
+  { \
+    (void)printf("ERROR: %s:%s:%d: ", __FILE__, __func__, __LINE__); \
+    (void)printf(__VA_ARGS__); \
+    fflush(stdout); \
+    fflush(stderr); \
```

Comment:
Needed? printf goes to stdout. I don't see stderr used. 
fprintf(stderr,)?

Reply:
According to @hihigupt, stream buffers need to be flushed prior to exit(), even in the case of a printf().  The approach to the samples is to always exit() on an error.
I don't explicitly see stderr used either, but was unsure if the MQTT library might be writing to it in the case of an error.

Reply:
should LOG_ERROR output the messages to stderr via fprintf?

Reply:
good idea. i'll check it out.